packages/router/src/lib/operations/prepare.ts

```diff
@@ -78,14 +80,88 @@ export const prepare = async (
     throw new SenderChainDataInvalid({ methodContext, requestContext });
   }
 
-  const inputDecimals = await txService.getDecimalsForAsset(invariantData.sendingChainId, invariantData.sendingAssetId);
+  let sendingChainIdx: number = -1;
+  let receivingChainIdx: number = -1;
+  let swapPoolIndex: number = -1;
+  config.swapPools.find((pool, index) => {
+    const existSwap =
+      pool.assets.find(
+        (a) =>
+          getAddress(a.assetId) === getAddress(invariantData.sendingAssetId) &&
+          a.chainId === invariantData.sendingChainId,
+      ) &&
+      pool.assets.find(
+        (a) =>
+          getAddress(a.assetId) === getAddress(invariantData.receivingAssetId) &&
+          a.chainId === invariantData.receivingChainId,
+      );
+
+    if (existSwap) {
+      sendingChainIdx = pool.assets.findIndex(
+        (a) =>
+          getAddress(a.assetId) === getAddress(invariantData.sendingAssetId) &&
+          a.chainId === invariantData.sendingChainId,
+      );
+
+      receivingChainIdx = pool.assets.findIndex(
+        (a) =>
+          getAddress(a.assetId) === getAddress(invariantData.receivingAssetId) &&
+          a.chainId === invariantData.receivingChainId,
+      );
+      swapPoolIndex = index;
+    }
+  });
```

Comment:
lets make a helper for this, we have the exact same (i believe) code in `newAuction`

Reply:
ok. will do!